File: Client/WarshipClient/Assets/CommandInterpreter.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;

public class CommandInterpreter
{
    public void treatIncomingCommand(Command c)
    {
        if(c.id == CommandManager.CommandID.GameStarts)
        {
            GameManager.instance.gameStarts();
        }
        else if(c.id == CommandManager.CommandID.NameSend)
        {
            StringCommand command = (StringCommand)c;
            GameManager.instance.registerPlayerName(command.param, command.data);
        }
        else
        {
            IntArrayCommand command = (IntArrayCommand)c;

            switch(c.id)
            {
                case CommandManager.CommandID.IDAttrib:
                    GameManager.instance.registerLocalPlayerID(c.param, command.data[0]);
                    break;
                case CommandManager.CommandID.RegisterOther:
                    GameManager.instance.registerPlayer(c.param, command.data[0]);
                    break;

                case CommandManager.CommandID.EndGame:
                    Debug.LogWarning(c.id + " Not Yet Implemented");
                    break;

                case CommandManager.CommandID.FireGrid:
                    Debug.LogWarning(c.id + " Not Yet Implemented");
                    break;

                case CommandManager.CommandID.FireResult:
                    Debug.LogWarning(c.id + " Not Yet Implemented");
                    break;

                case CommandManager.CommandID.TargetGrid:
                    Debug.LogWarning(c.id + " Not Yet Implemented");
                    break;

                case CommandManager.CommandID.ShipSunk:
                    Debug.LogWarning(c.id + " Not Yet Implemented");
                    break;
            }
        }
    }
}

File: Server/WarshipServer/CommandInterpreter.cpp
#include "CommandInterpreter.h"
#include "GameManager.h"

void CommandInterpreter::treatCommand(const Command* command)
{
	 if (command->id == Command::EndGame)
	{

	}
	else if (command->id == Command::NameSend)
	{
		StringCommand* c = (StringCommand*)command;
		gameManager->setupNewPlayer(c->socketID);
		gameManager->setPlayerName(c->socketID, c->data);
	}
	else
	{
		IntArrayCommand* c = (IntArrayCommand*)command;

		switch (command->id)
		{
			case Command::PlaceShip:
				gameManager->placePlayerShip(c->socketID, (Ship::ShipType)c->parameter, (Ship::Orientation)c->data[0], c->data[1]);
				break;

			case Command::GameStarts:
				gameManager->managePlayersReady(c->socketID, c->data[0]);
				break;

			case Command::TargetGrid:
				//managePlayerTargeting(int attackingPlayerID, int attackedPlayerID, int attackedNodeIndex)
				gameManager->managePlayerTargeting(c->socketID, c->parameter, c->data[0]);
				break;

			case Command::FireGrid:
				gameManager->managePlayerFire(c->socketID, c->data);
				break;

		}
	}

}
